File: MetaGeta/MetaGeta.DataStore/Database/MockDataMapper.vb
﻿Namespace Database
    Public Class MockDataMapper
        Implements IDataMapper

        Public Sub Initialize() Implements IDataMapper.Initialize

        End Sub

        Public Sub Close() Implements IDataMapper.Close

        End Sub

        Public Function GetTagOnAllFiles(ByVal dataStore As MGDataStore, ByVal tagName As String) _
            As IList(Of Tuple(Of MGTag, MGFile)) Implements IDataMapper.GetTagOnAllFiles
            Return New List(Of Tuple(Of MGTag, MGFile))
        End Function

        Public Function GetDataStores(ByVal owner As IDataStoreOwner) As IEnumerable(Of MGDataStore) Implements IDataMapper.GetDataStores
            Return New MGDataStore() {New MGDataStore(owner, Me) With {.Name = "Sample DataStore"}}
        End Function

        Public Function GetFiles(ByVal dataStore As MGDataStore) As IList(Of MGFile) Implements IDataMapper.GetFiles
            Return New MGFile() {}
        End Function

        Public Function GetGlobalSetting(ByVal settingName As String) As String Implements IDataMapper.GetGlobalSetting
            Return Nothing
        End Function

        Public Function GetPluginSetting(ByVal dataStore As MGDataStore, ByVal pluginID As Long, _
                                         ByVal settingName As String) As String Implements IDataMapper.GetPluginSetting
            Return String.Empty
        End Function

        Public Function ReadGlobalSettings() As IList(Of GlobalSetting) Implements IDataMapper.ReadGlobalSettings
            Return New GlobalSetting() {}
        End Function

        Public Sub RemoveDataStore(ByVal datastore As MGDataStore) Implements IDataMapper.RemoveDataStore

        End Sub

        Public Sub RemoveFiles(ByVal files As IEnumerable(Of MGFile), ByVal store As MGDataStore) _
            Implements IDataMapper.RemoveFiles

        End Sub

        Public Sub WriteDataStore(ByVal dataStore As MGDataStore) Implements IDataMapper.WriteDataStore

        End Sub

        Public Sub WriteGlobalSetting(ByVal settingName As String, ByVal settingValue As String) _
            Implements IDataMapper.WriteGlobalSetting

        End Sub

        Public Sub WriteNewDataStore(ByVal dataStore As MGDataStore) Implements IDataMapper.WriteNewDataStore

        End Sub

        Public Sub WriteNewFiles(ByVal files As IEnumerable(Of MGFile), ByVal dataStore As MGDataStore) _
            Implements IDataMapper.WriteNewFiles

        End Sub

        Public Sub WritePluginSetting(ByVal dataStore As MGDataStore, ByVal pluginID As Long, _
                                      ByVal settingName As String, ByVal settingValue As String) _
            Implements IDataMapper.WritePluginSetting

        End Sub

        Public Sub WriteFile(ByVal file As MGFile) Implements IDataMapper.WriteFile

        End Sub
    End Class
End Namespace

File: MetaGeta/MetaGeta.DataStore/Database/MockDataMapper.cs
// Copyright 2009 Will Thomas
// 
// This file is part of MetaGeta.
// 
// MetaGeta is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// MetaGeta is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with MetaGeta. If not, see <http://www.gnu.org/licenses/>.

#region

using System.Collections.Generic;
using MetaGeta.Utilities;
using System;

#endregion

namespace MetaGeta.DataStore.Database {
    public class MockDataMapper : IDataMapper {
        #region IDataMapper Members

        public void Initialize() {}


        public void Close() {}

        public IList<Tuple<MGTag, MGFile>> GetTagOnAllFiles(MGDataStore dataStore, string tagName) {
            return new List<Tuple<MGTag, MGFile>>();
        }

        public IEnumerable<MGDataStore> GetDataStores(IDataStoreOwner owner) {
            return new[] {new MGDataStore(owner, this) {Name = "Sample DataStore"}};
        }

        public IList<MGFile> GetFiles(MGDataStore dataStore) {
            return new MGFile[0];
        }

        public string GetGlobalSetting(string settingName) {
            return null;
        }

        public string GetPluginSetting(MGDataStore dataStore, long pluginID, string settingName) {
            return string.Empty;
        }

        public IList<GlobalSetting> ReadGlobalSettings() {
            return new GlobalSetting[0];
        }


        public void RemoveDataStore(MGDataStore datastore) {}


        public void RemoveFiles(IEnumerable<MGFile> files, MGDataStore store) {}


        public void WriteDataStore(MGDataStore dataStore) {}


        public void WriteGlobalSetting(string settingName, string settingValue) {}


        public void WriteNewDataStore(MGDataStore dataStore) {}


        public void WriteNewFiles(IEnumerable<MGFile> files, MGDataStore dataStore) {}


        public void WritePluginSetting(MGDataStore dataStore, long pluginID, string settingName, string settingValue) {}


        public void WriteFile(MGFile file) {}

        #endregion
    }
}